azext_edge/tests/edge/checks/int/helpers.py

```diff
@@ -0,0 +1,170 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from typing import Any, Dict, Optional, Tuple
+from azure.cli.core.azclierror import CLIInternalError
+from azext_edge.edge.providers.edge_api.base import EdgeResourceApi
+from ....helpers import find_extra_or_missing_names, get_kubectl_workload_items, run
+
+
+def assert_enumerate_resources(
+    post_deployment: dict,
+    description_name: str,
+    key_name: str,
+    resource_api: EdgeResourceApi,
+    resource_kinds: list,
+    present: bool = True
+):
+    # TODO: see if description_name and key_name can be combined for services (OPCUA)
+    key = f"enumerate{key_name}Api"
+    status = "success" if present else "skipped"
+    assert post_deployment[key]
+    assert post_deployment[key]["status"] == status
+    assert post_deployment[key]["description"] == f"Enumerate {description_name} API resources"
+
+    assert len(post_deployment[key]["targets"]) == 1
+    target_key = f"{resource_api.group}/{resource_api.version}"
+    assert target_key in post_deployment[key]["targets"]
+    evaluation = post_deployment[key]["targets"][target_key]["_all_"]
+    assert evaluation["conditions"] is None
+    assert evaluation["status"] == status
+    assert len(evaluation["evaluations"]) == 1
+    assert evaluation["evaluations"][0]["status"] == status
+    assert len(evaluation["evaluations"][0]["value"]) == len(resource_kinds)
+    for kind in evaluation["evaluations"][0]["value"]:
+        assert kind.lower() in resource_kinds
+
+
+# Used by Akri and OPCUA
+def assert_eval_core_service_runtime(
+    post_deployment: dict,
+    description_name: str,
+    pod_prefix: str,
+    resource_match: Optional[str] = None,
+):
+    assert post_deployment["evalCoreServiceRuntime"]
+    assert post_deployment["evalCoreServiceRuntime"]["description"] == f"Evaluate {description_name} core service"
+    overall_status = "success"
+    runtime_resource = post_deployment["evalCoreServiceRuntime"]["targets"]["coreServiceRuntimeResource"]
+    for namespace in runtime_resource.keys():
+        namespace_status = "success"
+        assert not runtime_resource[namespace]["conditions"]
+        evals = runtime_resource[namespace]["evaluations"]
+        kubectl_pods = get_kubectl_workload_items(
```

Comment:
will leave this for now but will need to make this into a helper function during mq tests